src/hats_import/catalog/map_reduce.py

```diff
@@ -263,6 +263,14 @@ def reduce_pixel_shards(
         healpix_pixel = HealpixPixel(destination_pixel_order, destination_pixel_number)
         destination_file = paths.pixel_catalog_file(output_path, healpix_pixel, npix_suffix=npix_suffix)
         if destination_file.exists():
+            merged_table = pq.read_table(destination_file)
+            rows_written = len(merged_table)
```

Comment:
1. Do we actually need to read the full file, or the metadata would be enough?
2. Should we use `hats`' file reader for the cases of the remote `destination_file`?